pacman/model/tags/tags.py

```diff
@@ -74,8 +74,10 @@ def add_ip_tag(self, ip_tag: IPTag, vertex: MachineVertex):
             raise PacmanInvalidParameterException(
                 "ip_tag", str(ip_tag), "Only add IP tags with this method.")
         existing_tag = None
-        if (ip_tag.board_address, ip_tag.tag) in self._ip_tags:
-            existing_tag = self._ip_tags[ip_tag.board_address, ip_tag.tag]
+        board_address = ip_tag.board_address
+        assert board_address is not None
```

Comment:
This seems unnecessary, or else should be handled; unless we know that the board_address is resolved elsewhere?

Reply:
fixed